### DISCUSSION TITLE
Add Intellisense for C# Markup resources.

### DISCUSSION THREAD

**Discussion by mcNets** [unoplatform/uno#15445] [Ideas]
I was trying/learning C# Markup language and I realized that now, when you create for example a new Brush, you get no Intellisense of the current static o themed resources.

```
this.Background(ThemeResource.Get<Brush>("ApplicationPageBackgroundThemeBrush"));
```

That makes it more difficult to use than the usual XAML code.

Maybe it would be possible,  perhaps using a source generator, to create some static Enum with the name of all existing resources in the app.  Something like:

```
this.Background(ThemeResource.Get<Brush>(AppBrushes.ApplicationPageBackgroundThemeBrush));
```

It would be great to have different Enums depending on the type of resource. (Brush, Color, BitmapImage, ...)

**Comment by dansiegel** [comment]
We currently have this mapped for those that are using the [Material Theme](https://www.nuget.org/packages/Uno.Themes.WinUI.Markup) you can read a little more on it in the [docs](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Markup/UnoThemes.html).

```cs
new Button()
    .Style(Theme.Button.Resources.Styles.Elevated)
```

Note that this also works with other values in the theme such as colors & brushes
